pyat/at/collective/wake_object.py

```diff
@@ -165,3 +165,49 @@ def reswall(self, wcomp, length, rvac, conduct, beta, yokoya_factor=1):
                                       length, rvac, conduct, beta)
         else:
             raise AtError('Invalid WakeComponent: {}'.format(wcomp))
+
+
+def res_object(srange, wakecomp, frequency, qfactor, rshunt, beta,
+              yokoya_factor=1):
+    """
+    Method to build a resonator wake object
+    """
+    wake = Wake(srange)
+    wakecomp = numpy.atleast_1d(wakecomp)
+    try:
+        frequency = numpy.broadcast_to(frequency, (len(wakecomp), ))
+        qfactor = numpy.broadcast_to(qfactor, (len(wakecomp), ))
+        rshunt = numpy.broadcast_to(rshunt, (len(wakecomp), ))
+        yokoya_factor = numpy.broadcast_to(yokoya_factor, (len(wakecomp), )) 
+    except ValueError:
+        raise AtError('Wake object inputs should be either scalars '
+                      'or with shape (len(wakecomp), )')
+    for wc, fr, qf, rs, yk in zip(wakecomp, frequency, qfactor, rshunt, yokoya_factor):
+        wake.add(WakeType.RESONATOR, wc, fr, qf, rs, beta, yokoya_factor=yk)
+    return wake
+
+
+def longres_object(srange, frequency, qfactor, rshunt, beta):
+    """
+    Method to build a longitudinal resonator wake object
+    """
```

Comment:
I would prefer to implement this as a static method of the `Wake` class rather than an independent function:
```python
@staticmethod
def long_resonator(srange, frequency, qfactor, rshunt, beta):
```
A longitudinal resonator object would be created with:
```python
long_res = Wake.long_resonator(srange, frequency, qfactor, rshunt, beta)
```
This cleans the global namespace and allows better names.
And the same for the `res_object` and `reswall_object`.